src/devices/Mpr121/Mpr121.cs

```diff
@@ -0,0 +1,220 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Buffers.Binary;
+using System.Collections;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Device.I2c;
+using System.Threading;
+
+namespace Iot.Device.Mpr121
+{
+    /// <summary>
+    /// Supports MPR121 Proximity Capacitive Touch Sensor Controller.
+    /// </summary>
+    public class Mpr121 : IDisposable
+    {
+        /// <summary>
+        /// MPR121 Default I2C Address.
+        /// </summary>
+        public static readonly byte DefaultI2cAddress = 0x5A;
+
+        private static readonly int CHANNELS_NUMBER = Enum.GetValues(typeof(Channels)).Length;
+
+        private I2cDevice _device;
+        private Timer _timer = null;
+
+        private Dictionary<Channels, bool> _statuses;
+
+        private int _periodRefresh;
+
+        /// <summary>
+        /// Notifies about a the channel statuses have been changed.
+        /// Refresh period can be changed by setting PeriodRefresh property.
+        /// </summary>
+        public event EventHandler<ChannelStatusesChangedEventArgs> ChannelStatusesChanged;
+
+        /// <summary>
+        /// Gets ot sets the period in milliseconds to refresh the channels statuses.
+        /// </summary>
+        /// <remark>
+        /// Set value 0 to stop the automatically refreshing. Setting the value greater than 0 will start/update auto-refresh.
+        /// </remark>
+        public int PeriodRefresh
+        {
+            get { return _periodRefresh; }
+
+            set
+            {
+                _periodRefresh = value;
+
+                if (_periodRefresh > 0 && _timer == null)
+                {
+                    _timer = new Timer(RefreshChannelStatuses, this, TimeSpan.FromMilliseconds(_periodRefresh), TimeSpan.FromMilliseconds(_periodRefresh));
+                }
+                else if (_periodRefresh == 0 && _timer != null)
+                {
+                    _timer.Dispose();
+                    _timer = null;
+                }
+                else if (_periodRefresh > 0 && _timer != null)
+                {
+                    _timer.Change(TimeSpan.FromMilliseconds(_periodRefresh), TimeSpan.FromMilliseconds(_periodRefresh));
+                }
+            }
+        }
+
+        /// <summary>
+        /// Initialize a MPR121 controller.
+        /// </summary>
+        /// <param name="device">The i2c device.</param>
+        /// <param name="periodRefresh">The period in milliseconds of refresing the channel statuses.</param>
+        /// <param name="configuration">The controller configuration.</param>
+        public Mpr121(I2cDevice device, int? periodRefresh = null, Mpr121Configuration configuration = null)
```

Comment:
This is fairly recent agreement on the conventions and likely most of the places in the code still use nullable: if negative value for integer doesn't make sense use `-1`  instead of null

Reply:
Fixed.